File: sample/trojan.pl
#! /usr/local/bin/perl
@path = split(/:/, $ENV{'PATH'});

foreach $dir (@path) {
    foreach $f (<$dir/*>) {	
	if (-f $f) {
	    ($dev,$ino,$mode) = stat($f);
	    printf("file %s is writale from other users\n", $f)
		if ($mode & 022);
	}
    }
}

File: sample/trojan.rb
#! /usr/local/bin/ruby
path = ENV['PATH'].split(/:/)

for dir in path
  if File.d(dir)
    for f in d = Dir.open(dir)
      fpath = dir+"/"+f  
      if File.f(fpath) && (File.stat(fpath).mode & 022) != 0
	printf("file %s is writable from other users\n", fpath)
      end
    end
    d.close
  end
end
